db/bigtable.go

```diff
@@ -886,6 +876,38 @@ func (bigtable *Bigtable) GetValidatorAttestationHistory(validators []uint64, st
 	if len(columnFilters) == 0 { // special case to retrieve data for all validators
 		filter = gcp_bigtable.FamilyFilter(ATTESTATIONS_FAMILY)
 	}
+
+	maxSlot := (endEpoch + 1) * utils.Config.Chain.Config.SlotsPerEpoch
+	err := bigtable.tableBeaconchain.ReadRows(ctx, ranges, func(r gcp_bigtable.Row) bool {
```

Comment:
I annoyingly had to have a second bigtable query here to get the highest relevant slot for the missed/orphaned slots.
This is a bug that is not directly related to the BIDS.
The issue is that until now we only checked slots for missed/orphaned until the end of the `endEpoch`.
However that is only the limit for the `AttesterSlot`, the `InclusionSlot` can go beyond that.
Getting the missed/orphaned map for all slots up to the current head takes way too long at least with the current `WHERE slot = ANY($1)` implementation.

Reply:
I rewrote the method to only have one bigtable call.